homeassistant/components/smtp/notify.py

```diff
@@ -143,14 +141,12 @@ def connection_is_valid(self):
                 self._server,
                 self._port,
             )
-            return False
 
         except (smtplib.SMTPAuthenticationError, ConnectionRefusedError):
             _LOGGER.exception(
                 "Login not possible. "
                 "Please check your setting and/or your credentials"
             )
-            return False
```

Comment:
If credentials are bad we should fail the platform setup since it's a non recoverable error.

Reply:
Good idea. I was going a bit more generic on this since my python skills are bad . I'll give it a go.